### ISSUE TITLE
Downcasting model object to another object

### ISSUE THREAD

**Issue by hipczor** [Raizlabs/DBFlow#204]
I have for example model class which looks like this

```java

@Table(databaseName = AppDatabase.NAME, value = TableNames.ATMS)
public class Atm extends BaseModel implements Parcelable {

    @Column(name = "atm_id", columnType = Column.PRIMARY_KEY)
    long atmId;

    @Column
    String bank;

    @Column
    String province;

    @Column
    String city;

    ....
}

```
and another object 

```java

public class CIty {
     
     String name;

     int count;

     ....
}

```
Now I want to achieve select statement which looks like this:
```sql
SELECT city as name, count(city) as count FROM atms WHERE province = '" + province
                + "' GROUP BY city ORDER BY city COLLATE LOCALIZED
```

So I need to use ```Select().from(Atm.class).where("conditions...")``` and save it into List<City> list.

Is there any way to downcast (extend smth) from one class into another using your library?

**Comment by agrosner** [comment]
Can you rephrase your question. What does smth mean?

And this seems like an interesting problem. For one it creates `count(city)` as its own column. I have a feature that allows `Join` statements to be represented as models in a different (non-final) branch. This may lead me to extend that functionality to include pretty much any statement.

**Comment by kamilzych** [comment]
@hipczor Right now it seems that the only way to do that is to use StringQuery. Am I right @agrosner?

```java
final List<City> list = new StringQuery<>(City.class, "SELECT city as name, count(city) as count FROM atms WHERE province = '" + province + "' GROUP BY city ORDER BY city COLLATE LOCALIZED").queryList();
```

assuming that you have class 'City' which looks like this:

```java
@Table(databaseName = AppDatabase.NAME, value = TableNames.ATMS)
public class City extends BaseModel {
    @Column(columnType = Column.PRIMARY_KEY)
    String name;

    @Column
    int count;

    public City() {
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getCount() {
        return count;
    }

    public void setCount(int count) {
        this.count = count;
    }
}
```

@agrosner BTW 'smth' stands for 'something' :smile:

**Comment by agrosner** [comment]
@hipczor I would avoid doing it that way. If you do it this way, `City` is treated as an actual table. I will make a way to ensure its not a table or created as such, but the `ModelAdapter` classes still work for it. This won't be ready until `2.0.0`. For now you'll have to manually parse data from the `Cursor` yourself.
